src/odrl/type/AccessGrant.test.ts

```diff
@@ -0,0 +1,399 @@
+    });
+
+    it("Rejects an ODRL-based Access Grant permission with an unknown operator", () => {
+      expect(
+        isCredentialAccessGrantOdrl({
+          ...mockAccessGrantOdrl,
+          credentialSubject: {
+            ...mockAccessGrantOdrl.credentialSubject,
+            permission: [
+              {
+                ...mockAccessGrantOdrl.credentialSubject.permission[0],
+                constraint: [
+                  {
+                    ...mockConstraint,
+                    operator: "Some unknown operator",
+                  },
+                ],
+              },
+            ],
+          },
+        })
+      ).toBe(false);
+    });
+
+    it("Rejects an ODRL-based Access Grant permission without a right operand", () => {
+      expect(
+        isCredentialAccessGrantOdrl({
+          ...mockAccessGrantOdrl,
+          credentialSubject: {
+            ...mockAccessGrantOdrl.credentialSubject,
+            permission: [
+              {
+                ...mockAccessGrantOdrl.credentialSubject.permission[0],
+                constraint: [
+                  {
+                    ...mockConstraint,
+                    rightOperand: undefined,
+                  },
+                ],
+              },
+            ],
+          },
+        })
+      ).toBe(false);
+    });
+  });
+
+  describe("Rejects ODRL-based Access Grant with invalid prohibitions", () => {
+    it("Rejects an ODRL-based Access Grant prohibitions with a malformed prohibition", () => {
+      expect(
+        isCredentialAccessGrantOdrl({
+          ...mockAccessGrantOdrl,
+          credentialSubject: {
+            ...mockAccessGrantOdrl.credentialSubject,
+            prohibition: "some invalid prohibition",
+          },
+        })
+      ).toBe(false);
+    });
+
+    it("Rejects an ODRL-based Access Grant prohibitions without a target", () => {
+      expect(
+        isCredentialAccessGrantOdrl({
+          ...mockAccessGrantOdrl,
+          credentialSubject: {
+            ...mockAccessGrantOdrl.credentialSubject,
+            prohibition: [
+              {
+                ...(
+                  mockAccessGrantOdrl.credentialSubject
+                    .prohibition as OdrlPermission[]
+                )[0],
+                target: undefined,
+              },
+            ],
+          },
+        })
+      ).toBe(false);
+    });
+
+    it("Rejects an ODRL-based Access Grant permission without an action array", () => {
+      expect(
+        isCredentialAccessGrantOdrl({
+          ...mockAccessGrantOdrl,
+          credentialSubject: {
+            ...mockAccessGrantOdrl.credentialSubject,
+            prohibition: [
+              {
+                ...(
+                  mockAccessGrantOdrl.credentialSubject
+                    .prohibition as OdrlPermission[]
+                )[0],
+                action: undefined,
+              },
+            ],
+          },
+        })
+      ).toBe(false);
+    });
```

Comment:
@NSeydoux this code could all be drier, `generateGrantMock({ action: undefined })` or something